Source/Core/InputCommon/DynamicInputTextureConfiguration.cpp

```diff
@@ -0,0 +1,261 @@
+  m_devicename = device.to_str();
+
+  SplitPath(json_file, &m_base_path, nullptr, nullptr);
+  if (!File::Exists(m_base_path + m_colorKeyImageName))
+  {
+    m_valid = false;
+    return;
+  }
+
+  const auto& emulated_controls_json = emulated_controls.get<picojson::object>();
+  for (auto& [emulated_control, color] : emulated_controls_json)
+  {
+    const int color_num = std::stoi(color.to_str().substr(1), 0, 16);
+    ColorKey color_key;
+    color_key.r = color_num / 0x10000;
+    color_key.g = (color_num / 0x100) % 0x100;
+    color_key.b = color_num % 0x100;
+
+    m_emulated_input_to_color.insert_or_assign(emulated_control, std::move(color_key));
+  }
+
+  const auto& host_controls_json = host_controls.get<picojson::object>();
+  for (auto& [host_control, image_name] : host_controls_json)
+  {
+    m_host_input_to_input_image.insert_or_assign(host_control, image_name.to_str());
+  }
+
+  // Optional fields
+  picojson::value& style = out.get("style");
+
+  if (style.is<std::string>())
+  {
+    m_style = style.to_str();
+  }
+
+  picojson::value& preserve_aspect_ratio = out.get("preserve_aspect_ratio");
+
+  if (preserve_aspect_ratio.is<bool>())
+  {
+    m_preserve_aspect_ratio = preserve_aspect_ratio.get<bool>();
+  }
+}
+
+void DynamicInputTextureConfiguration::GenerateTexture(IniFile::Section* sec)
+{
+  if (!m_valid)
+  {
+    return;
+  }
+
+  std::string device_name;
+  if (sec->Get("Device", &device_name) && device_name != m_devicename)
+  {
+    return;
+  }
+
+  auto rgb_to_hsv = [](u8 r, u8 g, u8 b) {
+    cv::Mat hsv;
+    cv::Mat rgb(1, 1, CV_8UC1, cv::Scalar(r, g, b));
+    cv::cvtColor(rgb, hsv, cv::COLOR_RGB2HSV);
+    return cv::Scalar(hsv.at<float>(0, 0), hsv.at<float>(0, 1), hsv.at<float>(0, 2));
+  };
+
+  // Load colorkey image copy
+  cv::Mat color_key_image = cv::imread(m_base_path + m_colorKeyImageName, cv::IMREAD_UNCHANGED);
+
+  cv::Mat hsv_color_key_image;
+  cv::cvtColor(color_key_image, hsv_color_key_image, cv::COLOR_BGR2RGB);
+
+  bool dirty = false;
+  for (auto& [emulated_key, color] : m_emulated_input_to_color)
+  {
+    std::string host_key;
+    if (!sec->Get(emulated_key, &host_key))
+    {
+      continue;
+    }
+
+    auto input_image_iter = m_host_input_to_input_image.find(host_key);
+    if (input_image_iter == m_host_input_to_input_image.end())
+    {
+      continue;
+    }
+
+    cv::Mat color_only_mask;
+    cv::inRange(hsv_color_key_image, cv::Scalar(color.r, color.g, color.b),
+                cv::Scalar(color.r, color.g, color.b), color_only_mask);
+
+    // get "regions" based on mask
+    std::vector<std::vector<cv::Point>> contours;
+    std::vector<cv::Vec4i> hierarchy;
+    cv::findContours(color_only_mask, contours, hierarchy, CV_RETR_TREE, CV_CHAIN_APPROX_SIMPLE,
+                     cv::Point(0, 0));
+
+    cv::Mat host_key_image =
+        cv::imread(m_base_path + input_image_iter->second, cv::IMREAD_UNCHANGED);
+
+    for (int i = 0; i < contours.size(); i++)
```

Comment:
This can be a ranged for loop (which also avoids a signed/unsigned mismatch here).